### PR TITLE
chore: add schema checking to the DataSource APIs

### PR DESCRIPTION
**Pull request by rsmayda**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: addons/addon-base-raas/packages/base-raas-services/lib/schema/register-data-source-account.json [hunk 1]
```diff
@@ -6,14 +6,17 @@
...
     },
     "description": {
       "type": "string",
-      "maxLength": 2048
+      "maxLength": 2048,
+      "pattern": "^([A-Za-z0-9-_ .,!?/#$;]+)$"
     },
     "contactInfo": {
       "type": "string",
+      "format": "email",
```
Comment (hatimkhan) [inline addons/addon-base-raas/packages/base-raas-services/lib/schema/register-data-source-account.json]: Can we use a pattern instead?  The contact info is supposed to be a free form, the user can input a combination of  phone numbers, names, addresses  and emails.

File: addons/addon-base-raas/packages/base-raas-services/lib/schema/register-study.json [hunk 2]
```diff
@@ -11,15 +11,17 @@
     },
     "name": {
       "type": "string",
-      "maxLength": 2048
+      "maxLength": 300,
+      "pattern": "^([A-Za-z0-9-_ ]+)$"
```
Comment (hatimkhan) [inline addons/addon-base-raas/packages/base-raas-services/lib/schema/register-study.json]: Do we want to handle cases where study names can contain umlauts and possibly other characters.

File: addons/addon-base-raas/packages/base-raas-services/lib/schema/register-study.json [hunk 3]
```diff
@@ -30,11 +32,13 @@
     "folder": {
       "type": "string",
       "minLength": 1,
-      "maxLength": 1000
+      "maxLength": 1000,
+      "pattern": "^([A-Za-z0-9-_ ]+)$"
```
Comment (hatimkhan) [inline addons/addon-base-raas/packages/base-raas-services/lib/schema/register-study.json]: We might want to account for more characters and umlauts.  including '@'.  In SWB default buckets, some folders contain '@'.

File: addons/addon-base-raas/packages/base-raas-services/lib/schema/update-data-source-account.json [hunk 4]
```diff
@@ -8,19 +8,23 @@
...
     },
     "description": {
       "type": "string",
-      "maxLength": 2048
+      "maxLength": 2048,
+      "pattern": "^([A-Za-z0-9-_ .,!?/#$;]+)$"
     },
     "contactInfo": {
       "type": "string",
+      "format": "email",
```
Comment (hatimkhan) [inline addons/addon-base-raas/packages/base-raas-services/lib/schema/update-data-source-account.json]: Can we use a pattern instead?  The contact info is supposed to be a free form, the user can input a combination of  phone numbers, names, addresses  and emails.

**Comment by hatimkhan** [comment]
Thank you Robert for updating the schema. In terms of Cross Site Scripting (XSS), in our design we don't depend on trying to prevent the user from entering characters like < or > or strings like <script>alert()</script>.  I agree it is a good idea to add one additional layer, we just need to make sure that we accounted for international study names.

In our current design we address XSS in two additional layers that we have in place:

- When we build Dynamo expressions and send data to the database, we use the AWS SDK document client which handles strings correctly.
- When we display anything in the browser, we use React which (unless you explicitly tell it to not do so) will always escape characters to avoid XSS.

File: addons/addon-base-raas/packages/base-raas-services/lib/schema/register-data-source-account.json (hunk 1 above)
Comment (rsmayda) [reply addons/addon-base-raas/packages/base-raas-services/lib/schema/register-data-source-account.json]: Why give so many options it seems confusing? Do we have a use-case that the contactInfo couldn't be an email?

**Review by hatimkhan — APPROVED** [review]

**Review by SanketD92 — APPROVED** [review]
LGTM

**Pull request merged**